python/cudf/cudf/core/column/column.py

```diff
@@ -990,6 +990,15 @@ def astype(self, dtype: Dtype, **kwargs) -> ColumnBase:
             np.object_,
             str,
         }:
+            if (
+                cudf.get_option("mode.pandas_compatible")
+                and not isinstance(self, cudf.core.column.StringColumn)
```

Comment:
Why should we allow string columns to be cast as `object`? I expected that _nothing_ should be cast-able to `object` since cudf doesn't truly support it.
```suggestion
```

Reply:
Agreed 👍 This however wouldn't be in-effect until we change our dtype to `np.dtype('str')` fully everywhere.